judge/pdf_problems.py

```diff
@@ -259,8 +269,48 @@ def _make(self, debug):
         self.log = self.proc.communicate()[0]
 
 
+class SeleniumPDFRender(BasePdfMaker):
+    success = False
+    template = {
+        'printBackground': True,
+        'displayHeaderFooter': True,
+        'headerTemplate': '<div></div>',
+        'footerTemplate': '<center style="margin: 0 auto; font-family: Segoe UI; font-size: 10px">' +
+                          gettext('Page [page] of [topage]').replace('[page]', '<span class="pageNumber"></span>')
+                                                            .replace('[topage]', '<span class="totalPages"></span>') +
+                          '</center>',
+    }
+
+    def _make(self, debug):
+        options = webdriver.ChromeOptions()
+        options.add_argument("--headless")
+        options.add_argument("--disable-gpu")
+        options.binary_location = settings.SELENIUM_CUSTOM_CHROME_PATH
+
+        browser = webdriver.Chrome(settings.SELENIUM_CHROMEDRIVER_PATH, options=options)
+        browser.get("file://" + os.path.abspath(os.path.join(self.dir, 'input.html')))
+
+        try:
+            WebDriverWait(browser, 15).until(EC.presence_of_element_located((By.CLASS_NAME, 'math-loaded')))
+        except TimeoutException:
+            logger.error('PDF math rendering timed out')
```

Comment:
This should be logged to `self.log` as well, as should all other exceptions.

Reply:
Added and formatted